ivadomed/adaptative.py

```diff
@@ -0,0 +1,695 @@
+import copy
+import os
+from os import path
+
+import h5py
+import numpy as np
+import pandas as pd
+from PIL import Image
+from bids_neuropoly import bids
+from medicaltorch import datasets as mt_datasets
+from tqdm import tqdm
+
+
+class Dataframe:
+    """
+    This class aims to create a dataset using an HDF5 file, which can be used by an adapative loader
+    to perform curriculum learning, Active Learning or any other strategy that needs to load samples in a specific way.
+    It works on RAM or on the fly and can be saved for later.
+    """
+
+    def __init__(self, hdf5, contrasts, path, target_suffix=None, roi_suffix=None,
+                 slices=False, dim=2):
```

Comment:
Not sure to understand in which case we would have `dim=2` and `slices=False`!?

Reply:
The parameter `slices` is misleading, it represents if the slices are filter or not (slice_filter_fn). So in 2D it is possible to filter or not the slices if gt or inputs are empty. I'll change the param name to slices_filter instead